taichi/transforms/detect_read_only.cpp

```diff
@@ -21,6 +22,43 @@ void detect_read_only_in_task(OffloadedStmt *offload) {
   }
 }
 
+class ExternalPtrAccessVisitor : public BasicStmtVisitor {
+ private:
+  std::unordered_map<int, int> &map;
```

Comment:
nit: please name these private members with a `_` suffix. Also, for consistency, let's place the member variables in the end of theclass definition.